pkg/cmd/server/kubernetes/network/network.go

```diff
@@ -232,17 +236,30 @@ func (c *NetworkConfig) RunProxy() {
 }
 
 // getNodeIP is copied from the upstream proxy config to retrieve the IP of a node.
-func getNodeIP(client kv1core.CoreV1Interface, hostname string) net.IP {
-	var nodeIP net.IP
-	node, err := client.Nodes().Get(hostname, metav1.GetOptions{})
-	if err != nil {
-		glog.Warningf("Failed to retrieve node info: %v", err)
-		return nil
+func getNodeIP(client kv1core.CoreV1Interface, hostname string) (net.IP, error) {
+	var node *v1.Node
+	var nodeErr error
+
+	// We may beat the thread that causes the node object to be created,
+	// so if we can't get it, then we need to wait.
+	// This will wait 2, 4, 8, ... 256 seconds, for a total of 8.5 mins
+	nodeWaitBackoff := utilwait.Backoff{
+		Duration: 2 * time.Second,
+		Factor:   2,
+		Steps:    8,
 	}
-	nodeIP, err = utilnode.GetNodeHostIP(node)
+	utilwait.ExponentialBackoff(nodeWaitBackoff, func() (bool, error) {
+		node, nodeErr = client.Nodes().Get(hostname, metav1.GetOptions{})
+		return nodeErr == nil, nil
+	})
+	if nodeErr != nil {
```

Comment:
But, don't I want to return `nodeErr` rather than the timeout error?  I think the error really is that we couldn't get the node object.  Returning `ErrWaitTimeout` just obscures the problem.